### DISCUSSION TITLE
How to let zeek only monitor interested  protocols ?

### DISCUSSION THREAD

**Discussion by cxdinter** [zeek/zeek#2419] [Help]
Hi,
    I know zeek is very powerful. It can analyze huge network flows. But, if I want to only monitor few protocols, how to config it ? 
    I have this idea, because I want to reduce  Zeek's  CPU load and memory usage during some stress testing cases (e.g. Nmap full ports scan). I am porting zeek into embedded system, the system resource is limited.
   For example, I only want to monitor SSH protocol, after configuration, zeek shall still can record the SSH login success/fail status.  , and don't need to analyze other type protocols.  I tried to change the BPF filters by 'default_capture_filter' in packeter_filter scripts, but I haven't found correct rules yet. After I modified the value of 'default_capture_filter', zeek cannot log SSH login success/fail behaviors. It seems zeek lost some packets. 
   
   Could you please guide me how to do the configuration ? Is it possible ? Thank you in advance.

**Comment by awelzel** [comment]
Hey @cxdinter - if you put the following into your `local.zeek` (or a separate script) all protocol analyzers will be disabled and only SSH is re-enabled - is this what you're looking for?

```
redef Analyzer::disable_all = T;

event zeek_init()
        {
        Analyzer::enable_analyzer(Analyzer::ANALYZER_SSH);
        }
```

> I tried to change the BPF filters by 'default_capture_filter' in packeter_filter scripts, but I haven't found correct rules yet.

If you use BPF (e.g for `tcp and port 22`), you may miss SSH connections on ports other than the well-known-ports. Take a peek at  `PacketFilter::exclude()` if that's okay for you.

> Could you please guide me how to do the configuration ? Is it possible ? Thank you in advance.

You can also post these questions on Slack (zeekorg.slack.com) or Discourse (community.zeek.org) for guidance.

**Reply by cxdinter** [reply]
> Hey @cxdinter - if you put the following into your `local.zeek` (or a separate script) all protocol analyzers will be disabled and only SSH is re-enabled - is this what you're looking for?
> 
> ```
> redef Analyzer::disable_all = T;
> 
> event zeek_init()
>         {
>         Analyzer::enable_analyzer(Analyzer::ANALYZER_SSH);
>         }
> ```
> 
>  > I tried to change the BPF filters by 'default_capture_filter' in packeter_filter scripts, but I haven't found correct rules yet.
> 
> If you use BPF (e.g for `tcp and port 22`), you may miss SSH connections on ports other than the well-known-ports. Take a peek at `PacketFilter::exclude()` if that's okay for you.
> 
> > Could you please guide me how to do the configuration ? Is it possible ? Thank you in advance.
> 
> You can also post these questions on Slack (zeekorg.slack.com) or Discourse (community.zeek.org) for guidance.

Hi awelzel,
    Thanks for your help !
    Your solution is good. But, at this direction, Zeek still will accept all network traffics data from network device, right? I mean, this solution will help to reduce the cpu load of zeek ?(e.g. nmap -A xx.xx.xx.xx -p- -Pn)
    You are right, if just filter packets by BPF rules, I will miss other SSH connections which not stay on port 22. Now I found one rule, it is working fine (yes, it will miss SSH connection which not deployed on port 22) :  const default_capture_filter = "(dst host 172.20.1.1 and (portrange 20-23 or port 80 or port 443 or port 8080 or port 13400)) or (src host 172.20.1.1) or not ip" &redef;

**Reply by awelzel** [reply]
> Your solution is good. But, at this direction, Zeek still will accept all network traffics data from network device, right?

Correct. You'll also continue to get basic conn.log entries for connections.

> I mean, this solution will help to reduce the cpu load of zeek ?(e.g. nmap -A xx.xx.xx.xx -p- -Pn)

It'll likely reduce CPU load, but also depends heavily on the kind of traffic Zeek is seeing. You should measure within the environment you'll be deploying Zeek and realistic traffic.

**Reply by cxdinter** [reply]
Thanks. I will try and update comments here.
